### ISSUE TITLE
NullCoalescingOperatorCanBeUsedInspection should not be applied for static class vars in methods

### ISSUE THREAD

**Issue by Denitz** [kalessil/phpinspectionsea#128]
```php
$s = isset(self::$test) ? self::$test : 'test';
$s = isset($classname::$test) ? $classname::$test : 'test';
```

Once change to NCO - we have error "Access to undeclared static property"

```php
$s = self::$test ?? 'test';
$s = $classname::$test ?? 'test';
```

Non-static class vars are OK.

**Comment by Denitz** [comment]
PHP bug  #74188 was fixed in PHP 7.0.19 and 7.1.5:
http://www.php.net/ChangeLog-7.php

Now null coalescing operator works for undeclared static properties.

**Comment by kalessil** [comment]
Thank you for an update @Denitz, reverting the patch from this ticket.

**Comment by Denitz** [comment]
Actually it should not be reverted but still active if PHP version is 7.0.0-7.0.18 or 7.1.0-7.1.4.

Probably it's possible to use version from 'Project Configuration > CLI Interpreter'.

**Comment by kalessil** [comment]
Hm, since this fixed IMO we can start reporting this again (we are relying on the language level, not interpreter).
Does e.g. your team plan to migrate to the new version withing 1 month?

**Comment by Denitz** [comment]
Hm... but PhpStorm has simple language levels like 7 and 7.1, hence we can't really check the exact version... Probably it's easier to remove this inspection and assume that the bug is fixed.

**Comment by kalessil** [comment]
Yes. Also an option to wait let say 3 months and start suggesting NCO for static fields later.
